pydantic/schema.py

```diff
@@ -234,6 +234,9 @@ def get_field_schema_validations(field: ModelField) -> Dict[str, Any]:
         f_schema['const'] = field.default
     if field.field_info.extra:
         f_schema.update(field.field_info.extra)
+    modify_schema = getattr(field.outer_type_, '__modify_schema__', None)
```

Comment:
maybe I'm being dumb, but can you explain why this is required?

Reply:
I didn't include it at first, but I found that the type schema and field schema are both setting the constraint properties. The dict update() method merges the output of the two, so to get rid of the props you need to call `__modify_schema__` in two places. Maybe it makes sense to call `__modify_schema__` only once, at the end of `field_schema`?

Reply:
if it's trivial to change it to only call this in one place and all tests pass, do that. Otherwise this is fine.

Reply:
Perhaps I am not really familiar with the codebase yet, but I don't think its trivial.